numba/cpython/builtins.py

```diff
@@ -550,8 +550,29 @@ def indval_min(indval1, indval2):
     if isinstance(indval1, IndexValueType) and \
        isinstance(indval2, IndexValueType):
         def min_impl(indval1, indval2):
-            if indval1.value > indval2.value:
+            # nan check
+            if indval1.value != indval1.value:
+                # nan check
+                if indval2.value != indval2.value:
```

Comment:
Should this use `np.isnan`?

Reply:
Up to you.  What I did should work for any kind of nan, not just np.nan but if that is the only kind possible then they are equivalent.

Reply:
IIRC `np.isnan` does a check for IEEE 754 defined `nan`, so should work for any conformant `nan`.
```pycon
In [1]: import numpy as np

In [2]: np.isnan(float('nan'))
Out[2]: True

In [3]: np.isnan(np.nan)
Out[3]: True

In [4]: qnan = (np.float64(float('nan')).view(np.uint64) | np.uint8(1)).view(np.float64)

In [5]: hex(qnan.view(np.uint64))
Out[5]: '0x7ff8000000000001'

In [6]: np.isnan(qnan)
Out[6]: True
```
my preference towards using `np.nan` is that it means that there's a single point in the code base to catch use of `fastmath` and "is this a nan", which means such an issue can be reported back to users from there.